modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/upgrade/v3_0_0/FragmentEntryLinkUpgradeProcess.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.fragment.internal.upgrade.v3_0_0;
+
+import com.liferay.fragment.entry.processor.constants.FragmentEntryProcessorConstants;
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alberto Sousa
+ */
+public class FragmentEntryLinkUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select editableValues from FragmentEntryLink where " +
+					"editableValues like '%\\\"collectionFieldId\\\":" +
+						"\\\"ObjectRelationship\\#%'");
+			ResultSet resultSet = preparedStatement1.executeQuery();
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection,
+					"update FragmentEntryLink set editableValues = ? where " +
+						"editableValues = ? ")) {
+
+			while (resultSet.next()) {
+				String editableValues = resultSet.getString("editableValues");
+
+				preparedStatement2.setString(
+					1, _replacePoundForUnderline(editableValues));
+
+				preparedStatement2.setString(2, editableValues);
+
+				preparedStatement2.addBatch();
+			}
```

Comment:
Adapt here since we will use `"update FragmentEntryLink set editableValues = ? where fragmentEntryLinkId = ? "`. You won't need to declare `editableValues` because it will only be used once.